pkg/client/cache/listers.go

```diff
@@ -135,7 +135,9 @@ func (s storeToNodeConditionLister) List() (nodes api.NodeList, err error) {
 		// Get the last condition of the required type
 		for _, cond := range node.Status.Conditions {
 			if cond.Type == s.conditionType {
-				nodeCondition = &cond
+				condCopy := cond
+				nodeCondition = &condCopy
```

Comment:
At the moment, we store only a single entry for a given condition type in node.Status.Conditions array. So break exists to ensure that we don't continue searching (do more work) even after finding our desired condition, because there cannot be one.

More over this fix is only for <=1.1. https://github.com/kubernetes/kubernetes/commit/a2570926da1b87856351c783c908c255042a9193#diff-53f28720440ab4b589ef2273346cecfcL139. In 1.2, this whole thing is going to change. https://github.com/kubernetes/kubernetes/pull/16179/files.

So it is pretty safe to apply this to 1.0.x even with break.